h/services/delete_user.py

```diff
@@ -23,17 +23,34 @@ def delete(self, user):
         message.
         """
 
-        if Group.created_by(self.request.db, user).count() > 0:
-            raise UserDeleteError('Cannot delete user who is a group creator.')
-
-        user.groups = []
+        # Check whether this user has created any groups that others have
+        # annotated in.
+        #
+        # We check for non-empty `group_ids` before querying the DB to avoid an
+        # expensive SQL query if `in_` is given an empty list (see
+        # https://stackoverflow.com/questions/23523147/)
+        group_ids = [g.pubid for g in user.groups]
+        if len(group_ids) > 0:
+            other_user_group_anns = self.request.db.query(Annotation) \
+                                                   .filter(Annotation.groupid.in_(group_ids),
+                                                           Annotation.userid != user.userid) \
+                                                   .count()
+            if other_user_group_anns > 0:
+                raise UserDeleteError('Other users have annotated in groups created by this user')
+
+        # Delete the user's annotations
         annotations = self.request.db.query(Annotation) \
                                      .filter_by(userid=user.userid)
         for annotation in annotations:
             storage.delete_annotation(self.request.db, annotation.id)
             event = AnnotationEvent(self.request, annotation.id, 'delete')
             self.request.notify_after_commit(event)
 
+        # Delete groups created by this user.
+        for group in user.groups:
```

Comment:
Same issue here as above, this will try to delete groups that the user is a _member_ of rather than ones they created.